server.js

```diff
@@ -43,14 +43,35 @@ app.get('/blocks', (request, response) => {
 
 app.get('/blocks/:index', (request, response) => {
     const index = request.params.index
-    const block = burgerNode.findBlockByIndex(index)
-    response.json(block) 
+    if(burgerNode.chain.length > index){
+        const block = burgerNode.findBlockByIndex(index)
+        response.json(block).status(200);
+    }else{
+        response.json({
+            "errorMsg": "Invalid block index"
+        }).status(404);
+    }
 })
 
 app.post('/mining/submit-mined-block', (request, response) => {
-    burgerNode.addMinedBlock(request.body);
+   let minedRes = burgerNode.addMinedBlock(request.body);
```

Comment:
We could convert this to a try-catch to make things easier.

try {
    burgerNode.addMinedBlock(request.body);
    ...
} catch (e) {
    response.json({
        "errorMsg": e.message.errorMsg
    }).status(e.message.responseCode);
}